addons/block_code/ui/blocks/utilities/parameter_input/parameter_input.gd

```diff
@@ -180,6 +196,8 @@ func get_snapped_block() -> Block:
 
 
 func _validate_and_submit_edit_text(line_edit: Node, type: Variant.Type):
+	if not editable:
+		return
```

Comment:
It seems weird to me that this is necessary when `_line_edit.editable` is set to match `editable` in `_set_editable`. I can see that if I focus (e.g.) the text field in the "log text" block, I can press `Enter` and trigger the callback that calls this function. But I can't find any way to actually change the value, so the check below also makes the function return early.

I suppose it's harmless to have this, and maybe there's some case I've overlooked which means it's actually necessary.

Reply:
@wjt you are right, this isn't necessary at all. I added this early return to avoid the emition of the `modified` signal at the end of this function. Nothing should be connected to that signal anyways for blocks in the picker. So do you think is better to remove the early returns from this patch?

Reply:
I think it's okay to keep it, I was just wondering what it's here for.

Reply:
(I think it is weird that signals like `LineEdit.text_submitted` can be emitted even when the field is not editable…)

Reply:
Yep, I think the same.